libcalico-go/lib/ipam/ipam.go

```diff
@@ -1564,8 +1575,8 @@ func (c ipamClient) RemoveIPAMHost(ctx context.Context, host string) error {
 	return errors.New("Max retries hit")
 }
 
-func (c ipamClient) hostBlockPairs(ctx context.Context, pool net.IPNet) (map[string]string, error) {
-	pairs := map[string]string{}
+func (c ipamClient) hostBlockPairs(ctx context.Context, pool net.IPNet) (map[string]model.AffinityConfig, error) {
```

Comment:
Whenever we change a function to be in terms of AffinityConfig instead of hostnames, we should be updating the name of the function to represent it's new meaning.

```suggestion
func (c ipamClient) affinityConfigsByBlock(ctx context.Context, pool net.IPNet) (map[string]model.AffinityConfig, error) {
```